analyses/cell-type-wilms-tumor-06/notebook/07_annotation_Across_Samples_exploration.Rmd

```diff
@@ -0,0 +1,586 @@
+                                      cell_type_df$has_cnv_score == "no CNV"]                          <- "normal"
+
+cell_type_df$first.level_annotation[  cell_type_df$compartment %in% c("endothelium", "immune") &
+                                      (cell_type_df$compartment.score > params$predicted.celltype.threshold |
+                                      cell_type_df$has_cnv_score == "no CNV") ]                        <- "normal"
+
+# Define cancer cells
+cell_type_df$first.level_annotation[  cell_type_df$compartment %in% c("fetal_nephron", "stroma") &
+                                      cell_type_df$has_cnv_score == "CNV" ]                            <- "cancer"
+
+```   
+
+Using this basic strategy, we identified `r table(cell_type_df$first.level_annotation)["cancer"]` _cancer_ cells, `r table(cell_type_df$first.level_annotation)["normal"]` _normal_ cells.
+Only `r table(cell_type_df$first.level_annotation)["unknown"]`cells remain _unknown_ 
+
+
+
+```{r fig.width=20, fig.height=20, out.width='100%', results='asis'}
+ggplot(cell_type_df, aes( x = umap.umap_1, y = umap.umap_2, color = first.level_annotation), shape = 19, size = 1)+
+  geom_point() +
+  facet_wrap(facets = ~sample_id, ncol = 5) +
+  theme_bw()  +
+  theme(text = element_text(size=22))
+
+```   
+
+
+### Second level annotation
+
+#### Normal cells
+
+- Normal cells from the `fetal nephron` compartment must be normal kidney cells.
+
+- Normal cells from the `stroma` compartment must be normal stroma cells.
+
+- Immune and endothelial cells have been already identified by label transfer.
+
+```{r fig.width=10, fig.height=10, out.width='100%', results='asis'}
+
+
+cell_type_df$second.level_annotation[  cell_type_df$compartment %in% c("fetal_nephron") &
+                                      cell_type_df$has_cnv_score == "no CNV"]                          <- "kidney"
+
+cell_type_df$second.level_annotation[  cell_type_df$compartment %in% c("stroma") &
+                                      cell_type_df$has_cnv_score == "no CNV"]                          <- "normal stroma"
+
+cell_type_df$second.level_annotation[  cell_type_df$compartment %in% c("endothelium") &
+                                      (cell_type_df$compartment.score > params$predicted.celltype.threshold |
+                                      cell_type_df$has_cnv_score == "no CNV") ]                        <- "endothelium"
+
+cell_type_df$second.level_annotation[  cell_type_df$compartment %in% c("immune") &
+                                      (cell_type_df$compartment.score > params$predicted.celltype.threshold |
+                                      cell_type_df$has_cnv_score == "no CNV") ]                        <- "immune"
+
+
+```   
+
+
+
+```{r fig.width=20, fig.height=20, out.width='100%', results='asis'}
+ggplot(cell_type_df[cell_type_df$first.level_annotation == "normal",], aes( x = umap.umap_1, y = umap.umap_2, color = second.level_annotation), shape = 19, size = 1)+
```

Comment:
For most of the cancer types, I guess we are used to have cancer cells that cluster separetly from normal cells. 

For Wilms tumor, I think this is however more complicated as cancer cells comprise _epithelial_, _stroma_ and _blastema_ cancer cells, that have more (transcriptional) similarities with their normal conterparts (i.e. normal kidney epithelium, normal reactive stroma) than between them.

For that reason, I expect for example _epithelial_ cancer and normal cells to be close, if not mixed, in the `umap` reduction. 

I then found easier to visualize cancer from normal cells separatly. But might be actually better to have to two plots side by side for each of the patient.

Reply:
I think what might help here is plotting a strategy I have used before to highlight cells in a UMAP - you can make all cells light gray, and then on top of that add a layer of your cells of interest that are colored. This way, you can clearly see the cells you care about, but still see the full context of the UMAP. 

Here's an example of how you might code something like this:

```
# data frame that only contains points of interest
subsetted_iris <- iris |>
  filter(Species == "versicolor")

ggplot(iris) + 
  aes(x = Sepal.Length, y =  Sepal.Width) + 
  geom_point(color = "gray") + 
  # add layer with points of interest colored
  geom_point(
    data = subsetted_iris, 
    aes(color = Species)
  ) 
```

Again though, this might be something to do later after the deadline!